subgraphs/compound-v3/src/sdk/manager.ts

```diff
@@ -0,0 +1,902 @@
+    amountUSD: BigDecimal,
+    newBalance: BigInt,
+    tokenPriceUSD: BigDecimal | null = null, // used for different borrow token
+    interestType: string | null = null
+  ): Repay | null {
+    const repayer = new AccountManager(
+      account,
+      this.market,
+      this.protocol,
+      this.event
+    );
+    if (repayer.isNewAccount()) {
+      this.protocol.cumulativeUniqueUsers += INT_ONE;
+      this.protocol.save();
+    }
+    repayer.subtractPosition(
+      newBalance,
+      PositionSide.BORROWER,
+      TransactionType.REPAY,
+      tokenPriceUSD ? tokenPriceUSD : this.market.inputTokenPriceUSD,
+      interestType
+    );
+    const positionID = repayer.getPositionID();
+    if (!positionID) {
+      log.error("[createRepay] positionID is null for market: {} account: {}", [
+        this.market.id.toHexString(),
+        account.toHexString(),
+      ]);
+      return null;
+    }
+
+    const repay = new Repay(
+      this.event.transaction.hash
+        .concatI32(this.event.logIndex.toI32())
+        .concatI32(Transaction.REPAY)
+    );
+    repay.hash = this.event.transaction.hash;
+    repay.nonce = this.event.transaction.nonce;
+    repay.logIndex = this.event.logIndex.toI32();
+    repay.gasPrice = this.event.transaction.gasPrice;
+    repay.gasUsed = this.event.receipt ? this.event.receipt!.gasUsed : null;
+    repay.gasLimit = this.event.transaction.gasLimit;
+    repay.blockNumber = this.event.block.number;
+    repay.timestamp = this.event.block.timestamp;
+    repay.account = account;
+    repay.market = this.market.id;
+    repay.position = positionID!;
+    repay.asset = asset;
+    repay.amount = amount;
+    repay.amountUSD = amountUSD;
+    repay.save();
+
+    this.snapshots.updateTransactionData(
+      TransactionType.REPAY,
+      amount,
+      amountUSD
+    );
+    this.snapshots.updateUsageData(TransactionType.REPAY, account);
+
+    return repay;
+  }
+
+  createLiquidate(
+    asset: Bytes,
+    liquidator: Address,
+    liquidatee: Address,
+    amount: BigInt,
+    amountUSD: BigDecimal,
+    profitUSD: BigDecimal,
+    newBalance: BigInt, // repaid token balance for liquidatee
+    interestType: string | null = null
+  ): Liquidate | null {
+    const liquidatorAccount = new AccountManager(
+      liquidator,
+      this.market,
+      this.protocol,
+      this.event
+    );
+    if (liquidatorAccount.isNewAccount()) {
+      this.protocol.cumulativeUniqueUsers += INT_ONE;
+      this.protocol.save();
+    }
+    liquidatorAccount.countLiquidate();
+    const liquidateeAccount = new AccountManager(
+      liquidatee,
+      this.market,
+      this.protocol,
+      this.event
+    );
+    liquidateeAccount.subtractPosition(
+      newBalance,
+      PositionSide.COLLATERAL,
+      TransactionType.LIQUIDATE,
+      this.market.inputTokenPriceUSD,
+      interestType
+    );
+    // Note:
+    //  - liquidatees are not considered users since they are not spending gas for the transaction
```

Comment:
Not considered users or active users (on liquidation)?

They should have already added to the cumulativeUniqueUsers since in order to be liquidated you have to have lent and borrowed first

Reply:
Yeah that is true, but just wanted to reinforce that we do not count (non-gas spenders) as cumuluative unique users